src/pythie_serving/table_wrapper.py

```diff
@@ -0,0 +1,90 @@
+
+    def __init__(self, *, logger: logging.Logger, model_server_config: model_server_config_pb2.ModelServerConfig):
+        self.logger = logger
+        self.table_map: Dict[str, Any] = {}
+        for table_config in model_server_config.model_config_list.config:
+            with open(os.path.join(table_config.base_path, "metadata.json"), "r") as f:
+                metadata = json.load(f)
+
+            with open(os.path.join(table_config.base_path, table_config.name) + ".csv", mode='r') as csvfile:
+                reader = csv.DictReader(csvfile)
+                raw_table = [row for row in reader]
+
+            # convert data type as csv reader only returns string
+            table_type_mapping = get_csv_type(metadata["data_type"])
+            table = [{feature: table_type_mapping[feature](feature_value)
+                      for feature, feature_value in row.items()} for row in raw_table]
+
+            self.table_map[table_config.name] = {
+                "table": table,
+                "feature_names": metadata["feature_names"],
+                "nb_features": len(metadata["feature_names"]),
+                "target_name": metadata["target_name"],
+                "data_type": metadata["data_type"]
+            }
+
+    def Predict(self, request: predict_pb2.PredictRequest, context: grpc.RpcContext):
+        table_name = request.model_spec.name
+        if table_name not in self.table_map:
+            raise PythieServingException(f'Unknown table: {table_name}. This pythie-serving instance can only '
+                                         f'serve one of the following: {",".join(self.table_map.keys())}')
+
+        table_dict = self.table_map[table_name]
+        table = table_dict["table"]
+        features_names = table_dict["feature_names"]
+        nb_features = table_dict["nb_features"]
+        target_name = table_dict["target_name"]
+        data_type = table_dict["data_type"]
+
+        for feature_name in features_names:
+            if feature_name not in request.inputs:
+                raise PythieServingException(f"{feature_name} not set in the predict request.")
+
+        nb_samples = request.inputs[features_names[0]].tensor_shape.dim[0].size
+        samples = np.empty((nb_samples, nb_features), object)
+
+        for feature_index, feature_name in enumerate(features_names):
+
+            if request.inputs[features_names[0]].tensor_shape.dim[0].size != nb_samples:
+                raise PythieServingException(f"{feature_name} has invalid length.")
+
+            nd_array = make_ndarray_from_tensor(request.inputs[feature_name])
+            if len(nd_array.shape) != 2 or nd_array.shape[1] != 1:
+                raise PythieServingException("All input vectors should be 1D tensor")
+
+            samples[:, feature_index] = nd_array.reshape(-1)
+
+        output = np.empty((nb_samples,))
+        for idx, sample in enumerate(samples):
+            pred = [record[target_name] for record in table if
```

Comment:
ok-ish but if you know the list of feature that you are going to use it would be better in `__init__` to construct a map with key = Tuple(*features_vales) and value = target and here to just compute the lookup key. Only problem is for the float but you could solve it by forcing to use Decimal instead of Float or have only int/str in the lookup key, which i think is a fair as prerequisite (i dont see a case where you would want to use a float as feature for a model based on CSV)

Reply:
Ok, I remove support for float (only str, int and bool are supported now) + construct the tap as a map